### PR TITLE
Feature - Refactor build process to remove hard-coded dependencies

### PR DESCRIPTION
**Pull request by priesgo**
* `build.py` completely refactored into new version `build2.py` which has the ability to manage by itself dependencies between packages, removing this burden from the models themselves and also maven.
  - Meaning that now models have not anymore relative imports, as dependencies are built together in a sandbox.
  - Only the snapshot was described into `build2.py`, it remains to be done on previous versions if everybody is happy.
* Backwards compatibility is maintained
  - `python build2.py` will just build all the models
  - while `python build2.py --skip-docs --version 4.3.0-SNAPSHOT` will just build `4.3.0-SNAPSHOT` without documentation
* Basic conversion operations implemented in a handy tool at `resources/GelModelsTools/gel_models_tools`
```
priesgo@londoner2:~/src/gel/GelReportModels$ python resources/GelModelsTools/gel_models_tools.py --help
usage: gel_models_tools.py <command> [<args>]

GEL models toolbox

positional arguments:
  command     Subcommand to run
              (idl2json|idl2avpr|json2java|idl2python|json2python|avpr2html)

optional arguments:
  -h, --help  show this help message and exit
priesgo@londoner2:~/src/gel/GelReportModels$ python resources/GelModelsTools/gel_models_tools.py idl2json --help
usage: gel_models_tools.py [-h] [--input INPUT] [--output OUTPUT]

Converts IDL to Avro JSON schema

optional arguments:
  -h, --help       show this help message and exit
  --input INPUT    Input folder containing *.avdl files
  --output OUTPUT  Output folder for the JSON schemas
```

Tests are OK and compilation of CVA too.
```
priesgo@londoner2:~/src/gel/GelReportModels$ python -m unittest discover
..................................................................
----------------------------------------------------------------------
Ran 66 tests in 2.115s

OK
```

### CODE REVIEW

**Comment by greglever** [comment]
@priesgo so I guess `__create_IDLs_build_folder` isn't quite running as expected in Docker ?

File: build2.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+
+def __create_IDLs_build_folder(packages):
+    """
+    Creates the build folder for a given set of packages
+    :param packages:
+    :return:
+    """
+    build_folder = os.path.join(IDL_FOLDER, "build")
+    distutils.dir_util.remove_tree(build_folder)
```
Comment (greglever) [inline build2.py]: this will fail if `build_folder` doesn't exist

**Comment by priesgo** [comment]
@Antonior26 please take a look I added the changes agreed this morning + the documentation of the whole process in the README. Legacy versions can now be built with maven as described in the README.

**Comment by Antonior26** [comment]
Looking quite good @priesgo

**Pull request merged**